test/BinaryParse.cpp

```diff
@@ -24,12 +24,421 @@
 // TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 // MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+#include "UnitSPIRV.h"
+
+#include "TestFixture.h"
+#include "gmock/gmock.h"
+
+// Returns true if two spv_parsed_operand_t values are equal.
+// To use this operator, this definition must appear in the same namespace
+// as spv_parsed_operand_t.
+static bool operator==(const spv_parsed_operand_t& a,
```

Comment:
Yes, but that doesn't compile, as alluded to in the blurb comment.

This is the error you get if you put the operator in the anonymous namespace instead:

stl_algobase.h: In instantiation of ‘static bool std::__equal<_BoolType>::equal(_II1, _II1, _II2) [with _II1 = const spv_parsed_operand_t*; _II2 = const spv_parsed_operand_t*; bool _BoolType = false]’:
...
/sr/include/c++/4.8/bits/stl_algobase.h:797:22: error: no match for ‘operator==’ (operand types are ‘const spv_parsed_operand_t’ and ‘const spv_parsed_operand_t’)

and about 630 more lines of errors showing you various things it tried.

The problem is that spv_parse_operand_t is declared outside of any namespace.  So an equality operator on it must be in the same namespace.